compiler/noirc_evaluator/src/ssa/opt/loop_invariant.rs

```diff
@@ -315,7 +314,11 @@ impl<'f> LoopInvariantContext<'f> {
     ) -> bool {
         if !matches!(
             binary.operator,
-            BinaryOp::Add { .. } | BinaryOp::Mul { .. } | BinaryOp::Sub { .. }
+            BinaryOp::Add { .. }
+                | BinaryOp::Mul { .. }
+                | BinaryOp::Sub { .. }
+                | BinaryOp::Div
+                | BinaryOp::Mod
         ) {
             return false;
         }
```

Comment:
I think we may want to make this a positive match rather than a negated one at this point. Also a quick comment explaining why we're rejecting these operators would be good.

Reply:
I decided to just remove the match.